File: android/src/main/java/com/mukha/andrei/plugins/device/secutiry/detect/DeviceSecurityDetectPlugin.kt
package com.mukha.andrei.plugins.device.secutiry.detect

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "DeviceSecurityDetect")
public class DeviceSecurityDetectPlugin : Plugin() {
    private val implementation = DeviceSecurityDetect()

    @PluginMethod
    public fun isJailBreakOrRooted(call: PluginCall) {
        val isDeviceRooted = implementation.isDeviceRooted()

        val ret = JSObject()
        ret.put("value", isDeviceRooted)
        call.resolve(ret)
    }

    @PluginMethod
    public fun pinCheck(call: PluginCall) {
        val ret = JSObject()
        ret.put("value", implementation.pinCheck(context))
        call.resolve(ret)
    }
}

File: ios/Sources/DeviceSecurityDetectPlugin/DeviceSecurityDetectPlugin.swift
import Foundation
import Capacitor

@objc(DeviceSecurityDetectPlugin)
public class DeviceSecurityDetectPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "DeviceSecurityDetectPlugin"
    public let jsName = "DeviceSecurityDetect"
    public let pluginMethods: [CAPPluginMethod] = [
        .async("isJailBreakOrRooted", DeviceSecurityDetectPlugin.isJailBreakOrRooted),
        .promise("pinCheck", DeviceSecurityDetectPlugin.pinCheck)
    ]
    private let implementation = DeviceSecurityDetect()

    /// Returns `{ value }`. One of the checks asks UIApplication whether it can open `cydia://`, and UIApplication is
    /// main-thread API, so the method runs on the main actor.
    @MainActor
    func isJailBreakOrRooted(_ call: CAPPluginCall) async -> JSObject {
        log("Checking if device is jailbroken from plugin")
        return [
            "value": implementation.isJailBreak()
        ]
    }

    func pinCheck(_ call: CAPPluginCall) {
        log("Checking PIN status from plugin")
        call.resolve([
            "value": implementation.pinCheck()
        ])
    }
}
